core/src/test/java/tech/tablesaw/table/TableSummaryTest.java

```diff
@@ -0,0 +1,68 @@
+package tech.tablesaw.table;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import org.junit.jupiter.api.Test;
+import tech.tablesaw.api.DoubleColumn;
+import tech.tablesaw.api.StringColumn;
+import tech.tablesaw.api.Table;
+
+public class TableSummaryTest {
+  @Test
+  public void summaryDoublesOnly() {
+    DoubleColumn value = DoubleColumn.create("value1").append(1.0).append(1.1).append(1.2);
+    DoubleColumn value2 = DoubleColumn.create("value2").append(2.0).append(2.1).append(2.2);
+
+    Table testTable = Table.create("Data");
+    testTable.addColumns(value);
+    testTable.addColumns(value2);
+    Table result = testTable.summary();
+    assertEquals(
+        "                            Data                             \n"
+            + " Summary   |         value1         |        value2         |\n"
+            + "-------------------------------------------------------------\n"
+            + "        n  |                     3  |                    3  |\n"
+            + "      sum  |                   3.3  |                  6.3  |\n"
+            + "     Mean  |                   1.1  |                  2.1  |\n"
+            + "      Min  |                     1  |                    2  |\n"
+            + "      Max  |                   1.2  |                  2.2  |\n"
+            + "    Range  |   0.19999999999999996  |  0.20000000000000018  |\n"
+            + " Variance  |  0.009999999999999995  |  0.01000000000000004  |\n"
+            + " Std. Dev  |   0.09999999999999998  |   0.1000000000000002  |",
+        result.print());
+  }
+
+  @Test
+  public void summaryMixedTypes() {
+    StringColumn label = StringColumn.create("label").append("row1").append("row2").append("row3");
+    DoubleColumn value = DoubleColumn.create("value1").append(1.0).append(1.1).append(1.2);
+    StringColumn label2 =
+        StringColumn.create("label2").append("row1").append("row5").append("row6");
+    DoubleColumn value2 = DoubleColumn.create("value2").append(2.0).append(2.1).append(2.2);
+
+    Table testTable = Table.create("Data");
+    testTable.addColumns(label);
+    testTable.addColumns(value);
+    testTable.addColumns(label2);
+    testTable.addColumns(value2);
+    Table result = testTable.summary();
+    assertEquals(
+        "                                       Data                                       \n"
+            + " Summary   |  label  |         value1         |  label2  |        value2         |\n"
+            + "----------------------------------------------------------------------------------\n"
+            + "     row1  |      1  |                        |       1  |                       |\n"
```

Comment:
I like the statistics! That's very cool!

I'm not sure about "row1", "row2", etc. It seems to me that it could make the table quite messy. It also doesn't appear that Pandas includes it in their output, so I think could also be unexpected or confusing for users coming from Python

Reply:
Thanks! I just checked the behaviour in pandas and it seems to be like this:

"Use the parameter, DataFrame.describe(include = 'all') to get a summary of all the columns when the dataframe has mixed column types. The default behavior is to only provide a summary for the numerical columns."

[DataFrame.describe()](https://pandas.pydata.org/pandas-docs/version/0.17.1/generated/pandas.DataFrame.describe.html)

Perhaps that is a better idea, to add a few overloaded methods that allow you to ask for summaries of specific column types, and to mirror that default behaviour.

Reply:
Even with `include = 'all'` the output includes only summary statistics:

```
>>> df.describe(include='all')
        categorical  numeric object
count            3      3.0      3
unique           3      NaN      3
top              f      NaN      c
freq             1      NaN      1
mean           NaN      2.0    NaN
std            NaN      1.0    NaN
min            NaN      1.0    NaN
25%            NaN      1.5    NaN
50%            NaN      2.0    NaN
75%            NaN      2.5    NaN
max            NaN      3.0    NaN
```

Reply:
Good point, those statistics seem to be a better representation of what a summary should be for a StringColumn, so I'd suggest we change the StringColumn.summary() method to return statistics. This is a breaking change but if someone was relying on it, they can easily switch to use StringColumn.countByCategory() directly

Reply:
I've tried that out in this latest set of changes. Once we finalize they feature I'll also update the tests so the don't assert on .print() and assert values

Reply:
We could also introduce a new method statistics() on Relation and Column and deprecate the summary() methods if that seems clearer.